### ISSUE TITLE
importing a fixture_union from subdir fails with KeyError

### ISSUE THREAD

**Issue by mskoenz** [smarie/python-pytest-cases#67]
Hi, first thx for the cool package!

### Situation
Have my fixtures and fixture union in a subdir and import it in my test file. It does not work, unless I import every fixture somehow part of the union.

```
test
├── my_test.py
└── subdir
    └── __init__.py
```

```
#__init__.py
from pytest_cases import pytest_fixture_plus, fixture_union

__all__ = ['fix_a', 'fix_b', 'fix_ab']

@pytest_fixture_plus
def fix_a():
    yield 'a'

@pytest_fixture_plus
def fix_b():
    yield 'b'

fix_ab = fixture_union('fix_ab', (fix_a, fix_b))  #tried 'fix_a', 'fix_b' too
```

```
#my_test.py
from subdir import fix_ab

def test_union(fix_ab):
    print(fix_ab)
```

### Fails
this fails upon calling pytest with:
```
=========================== test session starts ================================
platform darwin -- Python 3.7.0, pytest-5.3.0, py-1.8.0, pluggy-0.13.0
rootdir: .../test
plugins: cases-1.11.7
collected 0 items / 1 error

================================ ERRORS ========================================
___________________________ ERROR collecting my_test.py ______________________
.../site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.../site-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.../site-packages/pluggy/manager.py:86: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
.../site-packages/_pytest/python.py:240: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
.../site-packages/_pytest/python.py:410: in _genfunctions
    fixtures.add_funcarg_pseudo_fixture_def(self, metafunc, fm)
.../site-packages/_pytest/fixtures.py:118: in add_funcarg_pseudo_fixture_def
    if not metafunc._calls[0].funcargs:
.../site-packages/pytest_cases/plugin.py:693: in __getitem__
    return self.calls_list[item]
.../site-packages/pytest_cases/plugin.py:704: in calls_list
    self.create_call_list_from_pending_parametrizations()
.../site-packages/pytest_cases/plugin.py:740: in create_call_list_from_pending_parametrizations
    self._cleanup_calls_list(fix_closure_tree, calls, nodes, pending)
.../site-packages/pytest_cases/plugin.py:805: in _cleanup_calls_list
    fixdef = self.metafunc._arg2fixturedefs[fixture_name]
E   KeyError: 'fix_b'
!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.18s ===================================
```
### Ugly workaround
but if I `from subdir import *`it works since fix_a, fix_b are now present. It seems to resolve the fixtures on use, not on definition...

### Question
Is it possible to achieve what I'm trying here.
Apologies if I overlooked something in the doc.

### My setup
- OSX 10.15.1
- Python 3.7.0
- pytest==5.3.0
- pytest-cases==1.11.7

**Comment by smarie** [comment]
Nice one, thanks for reporting! I'll have a look in the upcoming days.

**Comment by smarie** [comment]
I was able to reproduce the bug. I'll have a look.

**Comment by mskoenz** [comment]
Great! I was just surprised is all, depending on how it is documented it doesn't have to be a bug :) I could imagine that changing that behavior has larger implications...

**Comment by smarie** [comment]
Apparently this is a known issue with pytest in general:

https://stackoverflow.com/questions/47402435/pytest-fixture-of-fixture-not-found

So I guess I wont "fix" anything but silently escape the KeyError so that you get the same error as everyone else ("fixture not found").

**Comment by mskoenz** [comment]
I tested the example and can confirm. I agree, it does not make sense to "fix" pytest, it would need to be a deliberate enhancement.
Thats a good idea, so googeling the error will lead people to pytest.
Thanks and feel free to close the issue.

**Comment by smarie** [comment]
perfect, thanks! fixed in 1.11.8
